recipes/systemc-components/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+from conans import ConanFile, tools, CMake
+from conan.errors import ConanInvalidConfiguration
+import functools
+import os
+import shutil
+
+required_conan_version = ">=1.43.0"
+
+class SystemcComponentsConan(ConanFile):
+    name = "systemc-components"
+    description = """A light weight productivity library for SystemC and TLM 2.0"""
+    homepage = "https://minres.github.io/SystemC-Components"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    topics = ("systemc", "modeling", "tlm", "scc")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "SC_WITH_PHASE_CALLBACKS": [True, False],
+        "SC_WITH_PHASE_CALLBACK_TRACING": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "SC_WITH_PHASE_CALLBACKS": False,
+        "SC_WITH_PHASE_CALLBACK_TRACING": False
+    }
+    generators = "cmake", "cmake_find_package_multi"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def validate(self):
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration(
+                f"{self.name} is not suppported on {self.settings.os}.")
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "7":
+            raise ConanInvalidConfiguration("GCC < version 7 is not supported")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.16.2")
+
+    @functools.lru_cache(1)
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["SC_WITH_PHASE_CALLBACKS"] = self.options.SC_WITH_PHASE_CALLBACKS
+        cmake.definitions["SC_WITH_PHASE_CALLBACK_TRACING"] = self.options.SC_WITH_PHASE_CALLBACK_TRACING
+        cmake.definitions["BUILD_SCC_DOCUMENTATION"] = False
+        cmake.configure(source_folder=self._source_subfolder)
```

Comment:
```suggestion
        cmake.configure()
```
If you point to source subfolder, it will skip your cmake wrapper.

Reply:
Could you please kindly explain what is the purpose of the cmake wrapper? I could not find it in the conan docu

Reply:
This one https://github.com/conan-io/conan-center-index/blob/master/docs/package/all/CMakeLists.txt.

It will be called by `cmake.configure()`, then it will run `conan_basic_setup` which will fit all `self.settings` with the cmake command. Otherwise, your Conan profile will not match with Cmake configuration and will produce an incorrect package.

Reply:
ok, I see. Thanks for the reveiw and explanation